File: 프로그래머스/lv0/120835. 진료 순서 정하기/진료 순서 정하기.js
function solution(emergency) {
    // emergency 배열의 원소들을 내림차순으로 나열
    let newEmergencyArray = [];
    let cnt = 0;
    while (cnt < emergency.length) {
        // 객체를 이용해서 원래 값, 위치 넣어줌
        newEmergencyArray.push({value: emergency[cnt], idx: cnt});
        cnt = cnt + 1;
    }
    // 내림차순으로 나열된 원소들의 순서를 정한 후 원래 위치로 이동
    newEmergencyArray.sort((a,b) => b.value - a.value);
    var answer = new Array(emergency.length);
    
    cnt = 0;
    while(cnt < newEmergencyArray.length) {
        // 배열에서 꺼낸 하나의 원소
        let item = newEmergencyArray[cnt]; 
        // 현재위치에서 중요도 정하기
        answer[item.idx] = cnt + 1; 
        cnt = cnt + 1;
    }
    return answer;
}

File: 프로그래머스/lv0/120835. 진료 순서 정하기/진료 순서 정하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] emergency) {
        // 깊은복사 : Arrays.copyOf(arr, length)
        // 얕은복사 : int[] arr = arr_b
        int[] answer = new int[emergency.length];
        int[] copy = Arrays.copyOf(emergency, emergency.length);
        
        Arrays.sort(copy);
        
        for(int i = 0; i < emergency.length; i++) {
            for(int j = 0; j < copy.length; j++) {
                if(emergency[i] == copy[j]) {
                    answer[i] = emergency.length - j;
                }
            }
        }
        return answer;
    }
}